include/revng/StackAnalysis/StackAnalysis.h

```diff
@@ -52,4 +62,169 @@ class StackAnalysis : public llvm::ModulePass {
   std::string TextRepresentation;
 };
 
+struct FuncSummary {
+  FuncSummary(
+    model::FunctionType::Values Type,
+    std::set<llvm::GlobalVariable *> ClobberedRegisters,
+    llvm::SmallSet<std::pair<llvm::CallInst *, model::FunctionEdgeType::Values>,
+                   4> Result,
+    std::optional<uint64_t> ElectedFSO,
+    llvm::Function *FakeFunction = nullptr) :
+    Type(Type),
+    ClobberedRegisters(ClobberedRegisters),
+    Result(Result),
+    ElectedFSO(ElectedFSO),
+    FakeFunction(FakeFunction) {}
+
+  FuncSummary() = delete;
+  FuncSummary(const FuncSummary &) = delete;
+  FuncSummary(FuncSummary &&) = default;
+  FuncSummary &operator=(const FuncSummary &) = delete;
+  FuncSummary &operator=(FuncSummary &&) = default;
+
+  model::FunctionType::Values Type;
+  std::set<llvm::GlobalVariable *> ClobberedRegisters;
+  llvm::SmallSet<std::pair<llvm::CallInst *, model::FunctionEdgeType::Values>,
+                 4>
+    Result;
+  std::optional<uint64_t> ElectedFSO;
+  llvm::Function *FakeFunction;
+
+  static bool compare(const FuncSummary &Old, const FuncSummary &New) {
+    if (New.Type == Old.Type)
+      return std::includes(Old.ClobberedRegisters.begin(),
+                           Old.ClobberedRegisters.end(),
+                           New.ClobberedRegisters.begin(),
+                           New.ClobberedRegisters.end());
+
+    return (New.Type <= Old.Type);
+  }
+
+  void dump() const debug_function { dump(dbg); }
+
+  template<typename T>
+  void dump(T &Output) const {
+    Output << "Dumping summary of the function. \n"
+           << "Type: " << Type << "\n"
+           << "Clobbered registers: ";
+
+    for (auto *Reg : ClobberedRegisters)
+      Output << ::getName(Reg);
+    Output << "\n";
+  }
+};
+
+class FunctionProperties {
```

Comment:
Is this the results cache?
Maybe call it `AnalysisResults` or `FunctionAnalysisResults`. Or something related to "cache".